addons/pos_restaurant/tests/test_frontend.py

```diff
@@ -269,3 +269,9 @@ def test_10_save_last_preparation_changes(self):
         self.start_tour("/pos/ui?config_id=%d" % self.pos_config.id, 'SaveLastPreparationChangesTour', login="pos_admin")
         self.assertTrue(self.pos_config.current_session_id.order_ids.last_order_preparation_change, "There should be a last order preparation change")
         self.assertTrue("Coca" in self.pos_config.current_session_id.order_ids.last_order_preparation_change, "The last order preparation change should contain 'Coca'")
+
+    def test_11_qrcode_ticket(self):
+        self.pos_config.write({'set_tip_after_payment': True, 'iface_tipproduct': True, 'tip_product_id': self.env.ref('point_of_sale.product_product_tip')})
```

Comment:
what does the `tip` have to do with this test?

Reply:
The first version of my fix was failing if tips where involved. But this is not required anymore because i use `finalized`